src/mscorlib/src/System/String.Searching.cs

```diff
@@ -96,8 +96,98 @@ public int IndexOfAny(char[] anyOf, int startIndex)
         }
 
         [Pure]
+        public int IndexOfAny(char[] anyOf, int startIndex, int count)
+        {
+            if (anyOf == null)
+            {
+                throw new ArgumentNullException(nameof(anyOf));
+            }
```

Comment:
> I wonder if it's worthwhile adding here

I think it is worthwhile to add it, and also Length == 0. Otherwise, the API has unexpected perf cliff.